src/Toolkit/Collection.php

```diff
@@ -1055,6 +1055,30 @@ public function values(): array
         return array_values($this->data);
     }
 
+    /**
+     * The when method only executes the given Closure when the first parameter
+     * is true. If the first parameter is false, the Closure will not be executed.
+     * You may pass another Closure as the third parameter to the when method.
+     * This Closure will execute if the first parameter evaluates as false
+     *
+     * @param mixed $condition
+     * @param Closure $callback
+     * @param Closure $fallback
+     * @return Collection
+     */
+    public function when($condition, Closure $callback, Closure $fallback = null)
+    {
+        if ($condition) {
+            return $callback->call($this, $condition) ?? $this;
```

Comment:
I'm not sure if we need the `?? $this` here. If the Closure decides that it wants to return `null`, that's what I'd expect to be the result. Or is there any use-case I'm missing @distantnative?

Same below.

Reply:
In fact this is for without return but need `Callable` instead of `Closure`?

````php
->when($sortBy, function ($sortBy) {
    $this->sortBy($sortBy);
});
````